server/tests/steps/sql_translator/test_rollup.py

```diff
@@ -0,0 +1,70 @@
+from unittest.mock import Mock
+
+import pytest
+
+from weaverbird.backends.sql_translator.metadata import ColumnMetadata
+from weaverbird.backends.sql_translator.steps.rollup import translate_rollup
+from weaverbird.pipeline.steps import RollupStep
+
+
+@pytest.fixture
+def sql_query_describer():
+    return Mock(return_value={'toto': 'int', 'raichu': 'str'})
+
+
+def test_rollup(query, sql_query_describer):
+    step = RollupStep(
+        name='rollup',
+        hierarchy=['CONTINENT', 'COUNTRY', 'CITY'],
+        aggregations=[
+            {'newcolumns': ['VALUE'], 'aggfunction': 'sum', 'columns': ['VALUE']},
+        ],
+    )
+    result = translate_rollup(step, query, index=1, sql_query_describer=sql_query_describer)
+    assert (
+        result.transformed_query
+        == '''WITH SELECT_STEP_0 AS (SELECT * FROM products), ROLLUP_STEP_1 AS (SELECT CONTINENT, COUNTRY, CITY, COALESCE(CITY, COUNTRY, CONTINENT) AS LABEL, CASE WHEN CITY IS NOT NULL THEN 'CITY' WHEN COUNTRY IS NOT NULL THEN 'COUNTRY' WHEN CONTINENT IS NOT NULL THEN 'CONTINENT' ELSE '' END AS LEVEL, CASE WHEN LEVEL = 'COUNTRY' THEN "CONTINENT" WHEN LEVEL = 'CITY' THEN "COUNTRY" ELSE NULL END AS PARENT, SUM(VALUE) AS VALUE FROM SELECT_STEP_0 GROUP BY ROLLUP(CONTINENT, COUNTRY, CITY) HAVING CONTINENT IS NOT NULL)'''
```

Comment:
Can we have this in multiline for a good readability ?

Reply:
:ok: